src/se/vidstige/jadb/JadbDevice.java

```diff
@@ -186,7 +186,7 @@ public void enableAdbOverTCP(int port) throws IOException, JadbException {
     public void push(InputStream source, long lastModified, int mode, RemoteFile remote) throws IOException, JadbException {
         try (Transport transport = getTransport()) {
             SyncTransport sync = transport.startSync();
-            sync.send("SEND", remote.getPath() + "," + Integer.toString(mode));
+            sync.send("SEND", remote.getPath() + "," + mode);
```

Comment:
`int`s will be converted to strings automatically, no? :thinking:

Reply:
Yes, that's what I'm suggesting.